### PR TITLE
Added meta project to manipulate all released components.

### PR DESCRIPTION
**Pull request by dblock**
Signed-off-by: dblock <dblock@amazon.com>

### Description

Added meta project to manipulate all released components. Moving from https://github.com/opensearch-project/opensearch-plugins/blob/main/META.md because this is where we do the releases. The template thus belongs here and also includes OpenSearch and OpenSearch Dashboards core repos. 
  
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: meta/.meta [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{
```
Comment (saratvemulapalli) [inline meta/.meta:1]: Why do we want to duplicate our component data? 
I see we already have https://github.com/opensearch-project/project-meta which takes care of this. 
Why not use it from opensearch-build for releases ?

Comment (dblock) [reply meta/.meta:1]: In meta we have all our repositories, not all components that are included in an OpenSearch release. I considered creating another folder in project-meta for just those, but effectively this list is the sum of OpenSearch + OpenSearch Dashboards manifests and is about what we release, which all lives in this repo.

**Review by peternied — APPROVED** [review]

File: meta/.meta (hunk 1 above)
Comment (saratvemulapalli) [reply meta/.meta:1]: Ok I would rather have one single source of truth and may be add parameters. 

Anyway, how do we plan on to keep this up to date?
I assume its manual and for a component owner this should be part of onboarding(ONBOARDING.md) to release workflow.

**Review by saratvemulapalli — APPROVED** [review]
Changes look good to me. 
Thanks @dblock !

Dropped a minor comment.

File: meta/.meta (hunk 1 above)
Comment (dblock) [reply meta/.meta:1]: I agree one source of truth is what we want, and the truth is here ;)

This is manual for now indeed. I'll think of some automation that could ensure that what's in manifests matches this list.

**Pull request merged**